lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -91,27 +91,19 @@ bool PagerandoBinning::runOnModule(Module &M) {
     auto I = Bins.lower_bound(FnSize);
     if (I == Bins.end()) {  // No bin with enough free space
       Bin = BinCount++;
-      FreeSpace = BinSize;
+      FreeSpace = BinSize - (FnSize % BinSize);
```

Comment:
Just realized: this may still not do exactly what we want. If `FnSize` is exactly a multiple of `BinSize`, `FreeSpace` will then equal `BinSize` above, therefore adding a new `BinSize`-sized bin. Instead, we might want to just not add a new bin (so that each function that exactly covers N pages just gets its own bin).